flink-core/src/main/java/org/apache/flink/metrics/reporter/JMXReporter.java

```diff
@@ -265,4 +329,72 @@ public Object getValue() {
 			return gauge.getValue();
 		}
 	}
+
+	/**
+	 * JMX Server implementation that JMX clients can connect to.
+	 *
+	 * Heavily based on j256 simplejmx project
+	 *
+	 * https://github.com/j256/simplejmx/blob/master/src/main/java/com/j256/simplejmx/server/JmxServer.java
+	 */
+	private static class JMXServer {
+		private int port;
+		private Registry rmiRegistry;
```

Comment:
dunno. It _looked_ important in the original, so i kept it. I only removed codepaths that wouldn't be used and the resulting unused fields etc.